### PR TITLE
create .NET Interactive extension package

### PR DESCRIPTION
**Pull request by colombod**
Add .NET Interactive extension pakcage

### CODE REVIEW

File: XPlot.Plotly.NET.Interactive/KernelExtension.fs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        newScript.AppendLine("""
+var renderPlotly = function() {
+    var xplotRequire = require.config({context:'xplot-3.0.1',paths:{plotly:'https://cdn.plot.ly/plotly-1.49.2.min'}}) || require;
+    xplotRequire(['plotly'], function(Plotly) {" """) |> ignore
+        newScript.AppendLine(script) |> ignore
+        newScript.AppendLine(@"});
+};"
+        ) |> ignore
+        newScript.AppendLine(JavascriptUtilities.GetCodeForEnsureRequireJs(Uri("https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js"), "renderPlotly")) |> ignore
+        newScript.ToString()
```
Comment (cartermp) [inline XPlot.Plotly.NET.Interactive/KernelExtension.fs]: Just for fun, this can now be an interpolated string since it's using the .NET 5 SDK

**Review by cartermp — CHANGES_REQUESTED** [review]
To get CI to pass I think we'll also have to change the tool manifest file and pull in the very latest version of paket and fake-cli

File: NuGet.config [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  </solution>
+  <packageSources>
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
+    <add key="dotnet-eng" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json" />
+    <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
+    <add key="dotnet3-dev" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3.1/nuget/v3/index.json" />
+    <add key="dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
+    <add key="MachineLearning" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/MachineLearning/nuget/v3/index.json" />
+    <add key="PSGallery" value="https://www.powershellgallery.com/api/v2/" />
```
Comment (daxian-dbw) [inline NuGet.config]: Unless you are going to pull PowerShell modules in this repo, this feed probably should be removed.

Comment (colombod) [reply NuGet.config]: I think I have to implement in this repo the code that the powershell kernel was doing to integrate the xplot types. What do you think?

Comment (daxian-dbw) [reply NuGet.config]: Yes, the `New-PlotyChart` cmdlet needs to be moved here. But the `PSGallery` is not needed for that. You will need to reference to [`PowerShellStandard.Library`](https://www.nuget.org/packages/PowerShellStandard.Library/5.1.0).

File: src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    [<Parameter(Position = 1)>]
+    member val Title = "" with get, set
+
+    [<Parameter(Position = 1)>]
+    member val Layout = Unchecked.defaultof<Layout.Layout> with get, set
+
+    override _.BeginProcessing() = ()
+
+    override _.ProcessRecord() = ()
```
Comment (daxian-dbw) [inline src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs]: Please do the same as the original implementation. This makes sure the piping behavior is as expected.
```
        private List<Trace> _traces;

        /// <summary>
        /// BeginProcessing override.
        /// </summary>
        protected override void BeginProcessing()
        {
            _traces = new List<Trace>();
        }
        
        /// <summary>
        /// ProcessRecord override.
        /// </summary>
        protected override void ProcessRecord()
        {
            _traces.AddRange(Trace);
        }
```

File: NuGet.config (hunk 2 above)
Comment (daxian-dbw) [reply NuGet.config]: This entry should be removed.

**Review by jonsequitur — APPROVED** [review]

File: src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs [hunk 4]
```diff
@@ -18,13 +22,16 @@ type NewPlotlyChartCommand() =
...
+    override this.BeginProcessing() = 
+        this._traces <- List<Trace>()
 
-    override _.ProcessRecord() = ()
+    override this.ProcessRecord() = 
+        this._traces.AddRange(this.Trace)
+        
 
     override this.EndProcessing() =
         let traces = if isNull this.Trace then [||] else this.Trace
```
Comment (daxian-dbw) [inline src/XPlot.Plotly.Interactive/NewPlotlyChartCommand .fs]: This line can be removed now

**Review by cartermp — APPROVED** [review]

**Pull request merged**